jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConfigFilesFacade.kt

```diff
@@ -188,6 +194,49 @@ class DefaultConfigFilesFacade(
         }
     }
 
+    override fun deleteSsoConnectionFromConfig(sessionName: String) {
+        val filePath = configPath
+        val lines = filePath.inputStreamIfExists()?.reader()?.readLines().orEmpty()
+        val ssoHeaderLine = lines.indexOfFirst { it.startsWith("[${SsoSessionConstants.SSO_SESSION_SECTION_NAME} $sessionName]") }
+        if (ssoHeaderLine == -1) return
+        val nextHeaderLine = lines.subList(ssoHeaderLine + 1, lines.size).indexOfFirst { it.startsWith("[") }
+        val endIndex = if (nextHeaderLine == -1) lines.size else ssoHeaderLine + nextHeaderLine + 1
+        val updatedArray = lines.subList(0, ssoHeaderLine) + lines.subList(endIndex, lines.size)
+        val profileHeaderLine = getCorrespondingSsoSessionProfilePosition(updatedArray, sessionName)
+        if (profileHeaderLine == -1) {
+            filePath.writeText(updatedArray.joinToString("\n"))
+        } else {
+            val nextHeaderLine2 = updatedArray.subList(profileHeaderLine + 1, updatedArray.size).indexOfFirst { it.startsWith("[") }
+            val endIndex2 = if (nextHeaderLine2 == -1) updatedArray.size else profileHeaderLine + nextHeaderLine2 + 1
+            filePath.writeText((updatedArray.subList(0, profileHeaderLine) + updatedArray.subList(endIndex2, updatedArray.size)).joinToString("\n"))
+        }
+        val applicationManager = ApplicationManager.getApplication()
+        if (applicationManager != null && !applicationManager.isUnitTestMode) {
+            FileDocumentManager.getInstance().saveAllDocuments()
+            ProfileWatcher.getInstance().forceRefresh()
+        }
+    }
+
+    private fun getCorrespondingSsoSessionProfilePosition(updatedArray: List<String>, sessionName: String): Int {
+        var content = updatedArray
+        var lineCnt = 0
+        while (content.size > 0) {
+            val pos = content.indexOfFirst { it.startsWith("[profile $sessionName-") }
```

Comment:
i don't think we're handling the case where there are profiles interspersed between the profiles we want to delete
```
[sso-session a]
...

[profile a-1]
sso_session = a
...

[profile something-else]
aws_access_key_id = ...
...

[profile a-2]
sso_session = a
...
```

Reply:
Is it possible to have the same sso-session associated with multiple profiles?

Reply:
This will be handled now